malcolm/modules/ADCore/parts/hdfwriterpart.py

```diff
@@ -557,6 +556,12 @@ def on_run(self, context: scanning.hooks.AContext) -> None:
             else:
                 return
 
+    @add_call_types
+    def on_post_run_armed(
+        self, context: scanning.hooks.AContext, steps_to_do: scanning.hooks.AStepsToDo,
+    ) -> None:
+        self.done_when_reaches += steps_to_do
```

Comment:
Why do we need this rather than sticking with calling on_seek on post run armed?

Reply:
on_seek fiddles with the UID and resets the arrayCounter. We don't want that. When we get into PostRunArmed with 100/300 done, we just want to say, OK now you need to get to 200/300, run again.